packages/gitbook/src/middleware.ts

```diff
@@ -335,7 +342,10 @@ function getInputURL(request: NextRequest): {
     // to determine the site to serve.
     const xGitbookSite = request.headers.get('x-gitbook-site-url');
     if (xGitbookSite) {
-        mode = 'proxy';
+        return {
+            mode: 'proxy',
+            url: new URL(xGitbookSite),
```

Comment:
From the `proxy` worker we send the entire URL that the user is trying to access, this is our `inputURL`